### PR TITLE
Add external delegation to support several NPU/GPU

### PR DESCRIPTION
**Pull request by ayakoakasaka**
Add VX delegation as an external delegation of TFLite. So that several NPU/GPU (from VeriSilicon, NXP, Amlogic) can be controlled via WASI-NN. Test Code can work with the X86 simulator.

### CODE REVIEW

File: core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp [hunk 1]
```diff
@@ -206,6 +210,17 @@ tensorflowlite_init_execution_context(void *tflite_ctx, graph g,
                 NN_ERR_PRINTF("Error when enabling GPU delegate.");
                 use_default = true;
             }
+#elif defined(WASI_NN_ENABLE_EXTERNAL_DELEGATE)
+            NN_WARN_PRINTF("external delegation enabled.");
+            auto options =
+                TfLiteExternalDelegateOptionsDefault(WASI_NN_EXT_DELEGATE_PATH);
+            auto *delegate = TfLiteExternalDelegateCreate(&options);
```
Comment (wenyongh) [inline core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp:217]: Had better check the return value? And same to do for L206?

**Comment by wenyongh** [comment]
@tonibofarull Could you help review this PR? Thanks.

File: core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp (hunk 1 above)
Comment (tonibofarull) [reply core/iwasm/libraries/wasi-nn/src/wasi_nn_tensorflowlite.cpp:217]: +1

**Review by wenyongh — COMMENTED** [review]
LGTM

**Review by tonibofarull — APPROVED** [review]

**Pull request merged**